### PR TITLE
Process status isXYZ methods removal

### PR DESCRIPTION
**Pull request by arkadius**

### CODE REVIEW

File: engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkDeploymentManager.scala [hunk 1]
```diff
@@ -58,7 +59,7 @@ abstract class FlinkDeploymentManager(modelData: BaseModelData, shouldVerifyBefo
       oldJob <- OptionT(checkOldJobStatus(processVersion))
       deploymentId <- OptionT.fromOption[Future](oldJob.deploymentId)
       //when it's failed we don't need savepoint...
-      if oldJob.status.isRunning || oldJob.status.isDuringDeploy
+      if Set(SimpleStateStatus.Running, SimpleStateStatus.DuringDeploy).contains(oldJob.status)
```
Comment (lciolecki) [inline engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkDeploymentManager.scala:62]: You declare this Set in the serval places. Let's move that to SimpleStateStatus.RunningStates = Set(SimpleStateStatus.Running, SimpleStateStatus.DuringDeploy). It provides an easier way to add/remove an option.

Comment (arkadius) [reply engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkDeploymentManager.scala:62]: I'll extract constant but only when you give me a good name for it. DuringDeploy certainly doesn't mean that scenario is running. Try again :-P

File: designer/deployment-manager-api/src/test/scala/pl/touk/nussknacker/engine/api/deployment/SimpleProcessStateSpec.scala [hunk 2]
```diff
@@ -15,19 +15,16 @@ class SimpleProcessStateSpec extends AnyFunSpec with Matchers with Inside {
...
 
   it ("scenario state should be running") {
     val state = createProcessState(SimpleStateStatus.Running)
-    state.status.isRunning shouldBe true
     state.allowedActions shouldBe List(ProcessActionType.Cancel, ProcessActionType.Pause, ProcessActionType.Deploy)
   }
 
   it ("scenario state should be finished") {
     val state = createProcessState(SimpleStateStatus.Finished)
-    state.status.isFinished shouldBe true
```
[the remark this answers is not in the document]
Comment (arkadius) [reply designer/deployment-manager-api/src/test/scala/pl/touk/nussknacker/engine/api/deployment/SimpleProcessStateSpec.scala:30]: So you want me to do in this test:
- create state wrapping status which is called finished 
- check if the name of this stattus is finished? 
- :-)  
What regression do you want to protect against? We don't check anywhere "FINISHED" string in production code, so IMO I shouldn't verify if this string is the same. It only concretes the code, It gives no positive impact that I can see.

File: designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/simple/SimpleStateStatus.scala [hunk 3]
```diff
@@ -14,11 +14,13 @@ object SimpleStateStatus {
   // Represents general problem.
   case class ProblemStateStatus(description: String, allowedActions: List[ProcessActionType] = defaultActions) extends StateStatus {
     override def name: StatusName = ProblemStateStatus.name
-    override def isFailed: Boolean = true
   }
 
   object ProblemStateStatus {
     val name: String = "PROBLEM"
```
[the remark this answers is not in the document]
Comment (arkadius) [reply designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/simple/SimpleStateStatus.scala:20]: Ehm, I'm not sure if I'm convicted that we should change the API for each small change. I agree with you that we should be consistent in naming constants, but should we change the small letters to the capital letter every time when we see this inconsistency? Maybe we should bundle all these kinds of changes and do this once? I know that we still don't have too big a community, but other mature projects care more about API changes.

File: extensions-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessState.scala [hunk 4]
```diff
@@ -52,43 +48,26 @@ object StateStatus {
...
-  }
-  def duringDeploy(statusName: StatusName): StateStatus = new StateStatus {
-    override def name: StatusName = statusName
-    override def isDuringDeploy: Boolean = true
-  }
+  def apply(statusName: StatusName): StateStatus = NoParamStateStatus(statusName)
 
-  def running(statusName: StatusName): StateStatus = new StateStatus {
-    override def name: StatusName = statusName
+  def running(statusName: StatusName): StateStatus = new NoParamStateStatus(statusName) {
```
[the remark this answers is not in the document]
Comment (arkadius) [reply extensions-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessState.scala:56]: As we talk, I'll fix it as well in a moment.

File: engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkDeploymentManager.scala (hunk 1 above)
Comment (lciolecki) [reply engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkDeploymentManager.scala:62]: ActiveStatutes? :)

File: designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/simple/SimpleStateStatus.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (lciolecki) [reply designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/simple/SimpleStateStatus.scala:20]: Ekhm.. ;)

File: designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/simple/SimpleStateStatus.scala [hunk 5]
```diff
@@ -14,11 +14,13 @@ object SimpleStateStatus {
...
+
+    def isProblemStatus(status: StateStatus): Boolean = status.name == name
```
[the remark this answers is not in the document]
Comment (arkadius) [reply designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/simple/SimpleStateStatus.scala:22]: Lowercase strings? Are we in java script? We are comparing identifiers, it is meant to be compared strictly.

File: designer/deployment-manager-api/src/test/scala/pl/touk/nussknacker/engine/api/deployment/SimpleProcessStateSpec.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (lciolecki) [reply designer/deployment-manager-api/src/test/scala/pl/touk/nussknacker/engine/api/deployment/SimpleProcessStateSpec.scala:30]: Ups, I shouldn't have done a review on Weekend.. :)

File: engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkDeploymentManager.scala (hunk 1 above)
Comment (lciolecki) [reply engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkDeploymentManager.scala:62]: In fact, you already named it: isFollowingDeployStatus ? :)

File: designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/simple/SimpleStateStatus.scala (hunk 5 above)
[the remark this answers is not in the document]
Comment (lciolecki) [reply designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/simple/SimpleStateStatus.scala:22]: Hmm, you're right. Let's leave it.

File: engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkDeploymentManager.scala (hunk 1 above)
Comment (arkadius) [reply engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkDeploymentManager.scala]: You won

**Pull request merged**